src/Arith.hs

```diff
@@ -32,37 +34,64 @@ toPostfix ast = error "toPostfix not implemented"
 -- "1 2 3 +" -> Nothing
 -- "1 2 + *" -> Nothing 
 fromPostfix :: String -> Maybe AST 
-fromPostfix input = error "fromPostfix not implemented"
+fromPostfix s = fromPostfix' s []
```

Comment:
Стилистическое замечание: if-выражения перегружают код, тут можно было сначала рассмотреть варианты, которые возвращают `Just что-то`, а все остальные случаи провалятся в `otherwise` ветку

Reply:
done